logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputDelegatorExt.java

```diff
@@ -39,14 +40,17 @@ public OutputDelegatorExt initialize(final ThreadContext context, final RubyHash
         final ExecutionContextExt executionContext,
         final OutputStrategyExt.OutputStrategyRegistryExt strategyRegistry) {
         this.outputClass = outputClass;
-        initMetrics(
-            args.op_aref(context, RubyString.newString(context.runtime, "id")).asJavaString(),
-            metric
-        );
+
+        String id = args.op_aref(context, RubyString.newString(context.runtime, "id")).asJavaString();
+        RubyString configRefKey = RubyString.newString(context.runtime, "config-ref");
+        String configRef = args.op_aref(context, configRefKey).asJavaString();
+        initMetrics(id, metric, configRef);
+
+        args.remove(configRefKey);
         strategy = (OutputStrategyExt.AbstractOutputStrategyExt) strategyRegistry.classFor(
             context, concurrency(context)
         ).newInstance(
-            context, new IRubyObject[]{outputClass, namespacedMetric, executionContext, args},
+            context, new IRubyObject[]{outputClass, namespacedMetric, executionContext, RubyUtil.RUBY.newString(configRef), args},
```

Comment:
unnecessary conversion -> you had it as a `RubyString` from args but converted to Java
just to convert back again :hammer: ... or maybe its a `RubySymbol` not a `RubyString`?
it could still be doing less work ... see ^^^